AnkiDroid/build.gradle

```
@@ -238,8 +238,22 @@ task installGitHook(type: Copy) {
 }
 tasks.getByPath(':AnkiDroid:preBuild').dependsOn installGitHook
 
+/**
+    The Kotlin compilation task names in Android projects contain the build variant names (those are combined from build type, product flavor and other settings and look like debug or releaseUnitTest -- the tasks are compileDebugKotlin and compileReleaseUnitTestKotlin respectively).
+    In kotlinOptions freeCompilerArgs(list of additional compiler arguments) is a attribute,
+    https://kotlinlang.org/docs/gradle.html#attributes-common-to-jvm-js-and-js-dce
+
+    There's no freeCompilerArgs in our kotlinCompile, we want to configure and to do that, we used freeCompilerArgs = ['-Xjvm-default=all'],
+    https://kotlinlang.org/docs/gradle.html#compiler-options
+
```

I'm sorry, but this comment is not illuminating at all - I had no idea what was going on here when I went to review it.
So I chased down the docs.

None of the part about task naming is important to understanding the build.gradle setting
Of course there was no freeCompilerArgs before, that's why we're adding it, but future reviewers won't care whether it was here before or not, they will want to know *what it does*

What it does is enable Kotlin<->Java interop for concrete implementations of functions inside Kotlin interfaces.

So the important link and comment is explaining that and linking to the interop documentation

```suggestion
Kotlin allows concrete function implementations inside interfaces.
For those to work when Kotlin compilation targets the JVM backend, you have to enable the interoperability via
'freeCompilerArgs' in your gradle file, and you have to choose one of the appropriate '-Xjvm-default' modes.

https://kotlinlang.org/docs/java-to-kotlin-interop.html#default-methods-in-interfaces

```

---

Thanks Mike for improving comments